src/gui/torrentoptionsdialog.cpp

```diff
@@ -378,11 +411,27 @@ void TorrentOptionsDialog::accept()
         BitTorrent::Torrent *torrent = session->findTorrent(id);
         if (!torrent) continue;
 
-        const QString savePath = m_ui->savePath->selectedPath();
         if (m_initialValues.autoTMM != m_ui->checkAutoTMM->checkState())
             torrent->setAutoTMMEnabled(m_ui->checkAutoTMM->isChecked());
-        if (!m_ui->checkAutoTMM->isChecked() && (m_initialValues.savePath != savePath))
-            torrent->move(Utils::Fs::expandPathAbs(savePath));
+
+        if (m_ui->checkAutoTMM->checkState() == Qt::Unchecked)
+        {
+            const QString savePath = m_ui->savePath->selectedPath();
+            if (m_initialValues.savePath != savePath)
+                torrent->setSavePath(Utils::Fs::expandPathAbs(savePath));
+
+            if (m_ui->checkUseDownloadPath->checkState() == Qt::Checked)
```

Comment:
I would give `m_ui->checkUseDownloadPath->checkState()` a local variable since it is used twice.

Reply:
> I would give `m_ui->checkUseDownloadPath->checkState()` a local variable since it is used twice.

Ok.